schema-registry/client/src/main/java/com/hortonworks/registries/schemaregistry/client/ISchemaRegistryClient.java

```diff
@@ -97,176 +90,51 @@
-         * @throws InvalidSchemaException      if the given schemaVersion is not valid
-         * @throws IncompatibleSchemaException if the given schemaVersion is incompatible according to the compatibility set.
-         * @throws SchemaNotFoundException if there is no schema metadata registered with the given {@code schemaName}
-         */
-    SchemaIdVersion addSchemaVersion(String schemaName, SchemaVersion schemaVersion) throws InvalidSchemaException, IncompatibleSchemaException, SchemaNotFoundException;
-
-    /**
-     * @param schemaFieldQuery {@link SchemaFieldQuery} instance to be run
-     * @return schema versions matching the fields specified in the query
-     */
-    Collection<SchemaVersionKey> findSchemasByFields(SchemaFieldQuery schemaFieldQuery);
-
-    /**
-     * @param schemaVersionKey key identifying a schema and a version
-     * @return {@link SchemaVersionInfo} for the given {@link SchemaVersionKey}
-     * @throws SchemaNotFoundException when there is no schema version exists with the given {@code schemaVersionKey}
-     */
-    SchemaVersionInfo getSchemaVersionInfo(SchemaVersionKey schemaVersionKey) throws SchemaNotFoundException;
-
-    /**
-     *
-     * @param schemaIdVersion Schema version's identifier
-     * @return {@link SchemaVersionInfo} for the given {@link SchemaIdVersion}
-     * @throws SchemaNotFoundException when there is no schema version exists with the given {@code SchemaIdVersion}
-     */
-    SchemaVersionInfo getSchemaVersionInfo(SchemaIdVersion schemaIdVersion) throws SchemaNotFoundException;
-
-    /**
-     * @param schemaName name identifying a schema
-     * @return latest version of the schema for the given schemaName
-     * @throws SchemaNotFoundException if there is no schema metadata registered with the given {@code schemaName}
-     */
-    SchemaVersionInfo getLatestSchemaVersionInfo(String schemaName) throws SchemaNotFoundException;
-
-    /**
-     * @param schemaName name identifying a schema
-     * @return all versions of the schemas for given schemaName
-     * @throws SchemaNotFoundException if there is no schema metadata registered with the given {@code schemaName}
-     */
-    Collection<SchemaVersionInfo> getAllVersions(String schemaName) throws SchemaNotFoundException;
-
-
     /**
      * @param schemaName   name identifying a schema
      * @param toSchemaText text representing the schema to be checked for compatibility
-     * @return true if the given {@code toSchemaText} is compatible with the latest version of the schema with id as {@code schemaName}.
+     * @return true if the given {@code toSchemaText} is compatible with all the versions of the schema with name as {@code schemaName}.
      * @throws SchemaNotFoundException if there is no schema metadata registered with the given {@code schemaName}
      */
     boolean isCompatibleWithAllVersions(String schemaName, String toSchemaText) throws SchemaNotFoundException;
```

Comment:
I think if the effort is to try align then where we can align be deprecating and old and adding a new it's worth it, as long as there's a pre agreed time the deprecated would be removed eg would revomed in next major +1 release